src/core/lib/security/security_connector/tls/tls_security_connector.cc

```diff
@@ -185,7 +185,7 @@ void TlsChannelSecurityConnector::check_peer(
    * server authorization check. */
   if (config != nullptr) {
     const tsi_peer_property* p =
-        tsi_peer_get_property_by_name(&peer, TSI_X509_PEM_CERT_PROPERTY);
+        tsi_peer_get_property_by_name(&peer, TSI_X509_PEM_CERT_CHAIN_PROPERTY);
```

Comment:
Instead of modifying the existing TSI peer property (`TSI_X509_PEM_CERT_PROPERTY`), I suggest to add a new property for `TSI_X509_PEM_CERT_CHAIN_PROPERTY` so that users who want to use a leaf certificate do not need to re-parse the certificate chain.

Reply:
Added.